src/go/rpk/pkg/api/admin/admin.go

```diff
@@ -18,197 +19,220 @@ import (
 	"fmt"
 	"io"
 	"io/ioutil"
+	"math/rand"
 	"net/http"
-	"strings"
+	"sync"
 	"time"
 
-	"github.com/Shopify/sarama"
 	"github.com/hashicorp/go-multierror"
-	log "github.com/sirupsen/logrus"
-)
-
-const (
-	usersEndpoint = "/v1/security/users"
-	httpPrefix    = "http://"
-	httpsPrefix   = "https://"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/net"
 )
 
+// AdminAPI is a client to interact with Redpanda's admin server.
 type AdminAPI struct {
 	urls   []string
 	client *http.Client
 }
 
-type newUser struct {
-	User      string `json:"username"`
-	Password  string `json:"password"`
-	Algorithm string `json:"algorithm"`
-}
-
+// NewAdminAPI returns client that talks to each of the input URLs.
+//
+// If tlsConfig is non-nil, the client talks to the URLs over https with the
+// given tls configuration.
 func NewAdminAPI(urls []string, tlsConfig *tls.Config) (*AdminAPI, error) {
-	adminUrls := make([]string, len(urls))
-	for i := 0; i < len(urls); i++ {
-		prefix := ""
-		url := urls[i]
-		// Go's http library requires that the URL have a protocol.
-		if !(strings.HasPrefix(url, httpPrefix) ||
-			strings.HasPrefix(url, httpsPrefix)) {
+	if len(urls) == 0 {
+		return nil, errors.New("at least one url is required for the admin api")
+	}
 
-			prefix = httpPrefix
+	a := &AdminAPI{
+		urls:   make([]string, len(urls)),
+		client: &http.Client{Timeout: 10 * time.Second},
+	}
+	if tlsConfig != nil {
+		a.client.Transport = &http.Transport{TLSClientConfig: tlsConfig}
+	}
 
+	for i, u := range urls {
+		scheme, host, err := net.ParseHostMaybeScheme(u)
+		if err != nil {
+			return nil, err
+		}
+		switch scheme {
+		case "", "http":
+			scheme = "http"
 			if tlsConfig != nil {
-				// If TLS will be enabled, use HTTPS as the protocol
-				prefix = httpsPrefix
+				scheme = "https"
 			}
-
-			url = strings.TrimRight(url, "/")
+		case "https":
+		default:
+			return nil, fmt.Errorf("unrecognized scheme %q in host %q", scheme, u)
 		}
-		adminUrls[i] = fmt.Sprintf("%s%s", prefix, url)
+		a.urls[i] = fmt.Sprintf("%s://%s", scheme, host)
```

Comment:
Similar to the logic before, this can override a scheme if a tls config is specified and the prior scheme was not present / was http. This also strips a trailing slash (from `ParseHostMaybeScheme`), so that the URLs we generate below can always have a leading slash.